blint/android.py

```diff
@@ -0,0 +1,277 @@
+        if "JAVA_TOOL_OPTIONS" in data:
+            parts = data.split("\n")
+            if parts and len(parts) > 0:
+                parts.pop(0)
+            return "\n".join(parts)
+    return None
+
+
+def collect_files_metadata(app_file):
+    """
+    Unzip the app and collect metadata
+    """
+    file_components = []
+    app_temp_dir = tempfile.mkdtemp(prefix="blint_android_app")
+    unzip_unsafe(app_file, app_temp_dir)
+    # Find and read all .version files
+    version_files = find_files(app_temp_dir, [".version"])
+    if version_files:
+        for vf in version_files:
+            file_name = os.path.basename(vf).removesuffix(".version")
+            rel_path = os.path.relpath(vf, app_temp_dir)
+            group = ""
+            name = ""
+            if "_" in file_name:
+                parts = file_name.split("_")
+                name = file_name
+                if parts and len(parts) == 2:
+                    group = parts[0]
+                    name = parts[-1]
+                else:
+                    name = name.replace("_", "-")
+                    # Patch the group name
+                    if name.startswith("kotlinx-"):
+                        group = "org.jetbrains.kotlinx"
+            with open(vf, encoding="utf-8") as fp:
+                version_data = fp.read().strip()
+                if name and version_data:
+                    if group:
+                        purl = f"pkg:maven/{group}/{name}@{version_data}"
+                    else:
+                        purl = f"pkg:maven/{name}@{version_data}"
+                    component = Component(
+                        type=Type.library,
+                        group=group,
+                        name=name,
+                        version=version_data,
+                        purl=purl,
+                        scope=Scope.required,
+                        evidence=ComponentEvidence(
+                            identity=Identity(
+                                field=FieldModel.purl,
+                                confidence=1,
+                                methods=[
+                                    Method(
+                                        technique=Technique.manifest_analysis,
+                                        value=rel_path,
+                                        confidence=1,
+                                    )
+                                ],
+                            )
+                        ),
+                        properties=[
+                            Property(name="internal:srcFile", value=rel_path),
+                            Property(name="internal:appFile", value=app_file),
+                        ],
+                    )
+                    component.bom_ref = RefType(purl)
+                    file_components.append(component)
+    # Parse all .so files
+    so_files = find_files(app_temp_dir, [".so"])
+    if so_files:
+        for sof in so_files:
```

Comment:
@prabhu Can we break this out into a separate function to improve maintainability?